include/revng/PipelineC/PipelineC.h

```diff
@@ -0,0 +1,416 @@
+#pragma once
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#ifdef __cplusplus
+#include <cstdint>
+
+#include "revng/Pipeline/Container.h"
+#include "revng/Pipeline/Kind.h"
+#include "revng/Pipeline/Runner.h"
+#include "revng/Pipeline/Step.h"
+#include "revng/Pipeline/Structure.h"
+#include "revng/Pipeline/Target.h"
+#include "revng/Pipes/PipelineManager.h"
+#else
+#include <stdint.h>
+#endif
+
+#include "stddef.h"
+#include "stdint.h"
+
+#ifdef C
+typedef struct rp_runner rp_runner;
+#else
+typedef Pipeline::PipelineManager rp_runner;
+#endif
+
+#ifdef C
+typedef struct rp_kind rp_kind;
+#else
+typedef const Pipeline::Kind rp_kind;
+#endif
+
+#ifdef C
+typedef struct rp_step rp_step;
+#else
+typedef Pipeline::Step rp_step;
+#endif
+
+#ifdef C
+typedef struct rp_container rp_container;
+#else
+typedef const Pipeline::ContainerBase rp_container;
+#endif
+
+#ifdef C
+typedef struct rp_target rp_target;
+#else
+typedef const Pipeline::Target rp_target;
+#endif
+
+#ifdef C
+typedef struct rp_runner_status rp_runner_status;
+#else
+typedef const Pipeline::Structure::State rp_runner_status;
+#endif
+
+#ifdef C
+typedef struct rp_step_status rp_step_status;
+#else
+typedef const Pipeline::ContainersStatus rp_step_status;
+#endif
+
+#ifdef C
+typedef struct rp_container_status rp_container_status;
+#else
+typedef const Pipeline::TargetsList rp_container_status;
+#endif
+
+#ifdef CPP
+extern "C" {
+#endif
+
+/**
+ * Libraries rule, ever returned pointer is owned by the library and must not be
+ * deleated, except for those functions that contains the word "create". The
+ * objects returned by the functions named "create" are owned by the caller and
+ * must be destroyed invoking the appropriated function called rp_TYPE_destroy.
+ *
+ */
```

Comment:
Use doxygen for doc `:)`

Reply:
/** is a doxygen accepted "opener"

do we need to use /// instead everywhere?

Reply:
Ah didn't know that! That works.